src/storage/hp/primera/restapi/mode/listdisks.pm

```diff
@@ -0,0 +1,111 @@
+#
+# Copyright 2024 Centreon (http://www.centreon.com/)
+#
+# Centreon is a full-fledged industry-strength solution that meets
+# the needs in IT infrastructure and application monitoring for
+# service performance.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package storage::hp::primera::restapi::mode::listdisks;
+
+use strict;
+use warnings;
+
+use base qw(centreon::plugins::mode);
+
+sub new {
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options);
+    bless $self, $class;
+
+    return $self;
+}
+
+sub check_options {
+    my ($self, %options) = @_;
+    $self->SUPER::init(%options);
+}
+
+sub run {
+    my ($self, %options) = @_;
+
+    my $response = $options{custom}->request_api( endpoint => '/api/v1/disks' );
+    my $disks = $response->{members};
+
+    $self->{disks} = [];
+
+    for my $disk (@{$disks}) {
+        push @{$self->{disks}}, {
+                id           => $disk->{id},
+                position     => $disk->{position},
+                size         => $disk->{totalSizeMiB},
+                manufacturer => $disk->{manufacturer},
+                model        => $disk->{model},
+                serial       => $disk->{serialNumber}
+        };
+        $self->{output}->output_add(
+            long_msg => sprintf(
+                "[id: %s][position: %s][size: %s][manufacturer: %s][model: %s][serial: %s]",
+                $disk->{id},
+                $disk->{position},
+                $disk->{totalSizeMiB},
+                $disk->{manufacturer},
+                $disk->{model},
+                $disk->{serialNumber}
+            )
+        );
+    }
+
+    if (!defined($options{disco_show})) {
```

Comment:
Strange condition, what happen if the client set disco-show and the disco-format option ?